x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/planner/QueryTranslator.java

```diff
@@ -108,6 +110,15 @@ public static Query doTranslate(ScalarFunction f, TranslatorHandler handler) {
                     return new TermsQuery(f.source(), targetFieldName, set);
                 }
             }
+            if (f instanceof StringContains) {
+                StringContains sc = (StringContains) f;
+                if (sc.isCaseSensitive() && sc.string() instanceof FieldAttribute && sc.substring().foldable()) {
+                    String targetFieldName = handler.nameOf(((FieldAttribute) sc.string()).exactAttribute());
+                    String substring = (String) sc.substring().fold();
+
+                    return new WildcardQuery(f.source(), targetFieldName, "*" + substring + "*");
+                }
+            }
```

Comment:
Wouldn't this work better in ExpressionTranslators#Scalars where `StartsWith` is handled?

Reply:
Speaking of, having a rule for `CaseSensitiveScalarFunction` might work better since it handle the `if (isCaseSensitive && FieldAttribute && foldable) so subclasses would only look at the name and the query while reusing the infrastructure.

Reply:
`StartsWith` is special because is common to SQL and EQL while `stringContains`, `endsWith` belong only to EQL. Agree with a separate rule, though.